src/TheThingsNetwork.cpp

```diff
@@ -436,6 +442,18 @@ void TheThingsNetwork::reset(bool adr)
     sendMacSet(MAC_ADR, "off");
   }
   this->adr = adr;
+  this->needsHardReset = false;
+}
+
+void TheThingsNetwork::resetHard(uint8_t resetPin){
+  digitalWrite(resetPin, LOW);
+  unsigned long time_now = millis();
+  int period = 1000;
+  // Non I/O blocking wait function
+  while(millis() < time_now + period)
+  {
+  }
+  digitalWrite(resetPin, HIGH);
```

Comment:
```
  digitalWrite(resetPin, LOW);
  delay(1000);
  digitalWrite(resetPin, HIGH);
```

We need to use delay, and not write our own delay loop. This is so that this code won't crash the wifi on the ESP8266. Alternatively one can add a `yield()` in the body of the while, but not all platforms has the yield() function. So better use `delay()`.

Reply:
I have not worked with delay functions on ESP8266, but i would assume that using delay function it would block the wifi operation, just as the delay function on a "normal" arduino blocks the interrupt. 
Can you explain why the wifi will crash using this delay loop?

Reply:
"the delay function on a "normal" arduino blocks the interrupt" - I'm not sure that this is true. A delay inside an ISR will prevent further interrupts to trigger, but this is not true in the normal execution path. So a delay in normal code will be interrupted just like any other code.

But let me reiterate this:
Do not write custom implementations where perfectly working functions are given by the platform. Different platforms handle things differently lower down, and the abstraction that Arduino gives us should be used to prevent thing from going wrong. The ESP8266 example is just one of many possible situations where a custom blocking delay loop will cause issues with lower down system services.

On https://learn.sparkfun.com/tutorials/esp8266-thing-hookup-guide/using-the-arduino-addon, read the paragraph under Yielding.

Reply:
Thanks, it is pretty clear to me now. I will update my PR.